packages/cypress/src/add-commands/login.ts

```diff
@@ -0,0 +1,290 @@
+import { v4 as uuidv4 } from 'uuid';
+import { buildOidcScope } from '@commercetools-frontend/application-shell/ssr';
+import type { ApplicationRuntimeConfig } from '@commercetools-frontend/application-config';
+import { STORAGE_KEYS, OIDC_RESPONSE_TYPES } from '../constants';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+declare const Cypress: any;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+declare const cy: any;
+
+export type LoginCredentials = {
+  /**
+   * The user email.
+   * Defaults to `Cypress.env('LOGIN_EMAIL') ?? Cypress.env('LOGIN_USER')`.
+   */
+  email: string;
+  /**
+   * The user password.
+   * Defaults to `Cypress.env('LOGIN_PASSWORD')`.
+   */
+  password: string;
+};
+
+export type CommandLoginOptions = {
+  /**
+   * The application entry point URI path. This value is used to identify
+   * the correct application config.
+   */
+  entryPointUriPath: string;
+  /**
+   * Pass a list of dotfiles that must be loaded when the custom-application-config.json
+   * is loaded (in case you are using environment placeholder).
+   * By default the following dotfiles are loaded: `.env` and `.env.local`.
+   * You can also define the values using paths relative to the application folder.
+   */
+  dotfiles?: string[];
+  /**
+   * Called before your page has loaded all of its resources.
+   * Use this as a chance to interact for example with the browser storage.
+   */
+  onBeforeLoad?: (win: Window) => void;
+  /**
+   * If defined, visit this route after login.
+   */
+  initialRoute?: string;
+  /**
+   * The project key to access in the user session. The session token is valid for one project key at a time.
+   * Defaults to `Cypress.env('PROJECT_KEY')`.
+   */
+  projectKey?: string;
+  /**
+   * The user login credentials.
+   */
+  login?: LoginCredentials;
+};
+// Alias for backwards compatibility
+export type CommandLoginByOidcOptions = CommandLoginOptions;
+
+function loginByForm(commandOptions: CommandLoginOptions) {
+  if (isLocalhost()) {
+    throw new Error(
+      `At the moment, the "loginByForm" command only works when testing a Merchant Center production URL. Testing an application running on localhost is not supported due to issues with "cy.origin".`
```

Comment:
**nit**: the message seems a little confusing to me as it says we can't test an application running locally.
```suggestion
      `At the moment, the "loginByForm" command only works when testing a Merchant Center production URL. Using form login in an application running on localhost is not supported due to issues with "cy.origin".`
```